Industry and friends

umbobabo opened the pull request:
### Description
This PR port the field industry, job title and position from HBS to JSX.

It is composed of 2 separated commits:
1. Partials (Changes necessary to fix error in partials that React doesn't like)
2. Code and tests

[Ticket](https://trello.com/c/1mVk1BEQ/92-personal-details-port-industry-job-title-position-to-jsx)

### Reminder
Have you completed these common tasks (remove those that don't apply)?

- [ ] **Documentation** updated or created
- [x] **Tests** written for new or updated for existing functionality
- [x] **Demos** updated to use this change
- [ ] **Accessibility** checked for screen readers and contrast
- [ ] **Design Review** ran past the designer 
- [ ] **Product Review** ran past the product owner

ludrob commented on components/industry.jsx:
```diff
@@ -0,0 +1,56 @@
+				return <option key={code} value={code}>{description}</option>;
+			})}
+		</select >
+		<div className="o-forms__errortext" >Please select your company‘s industry</div >
```
```suggestion
		<div className="o-forms__errortext" >Please select your company's industry</div >
```

umbobabo replied:
I cannot change this without changing it in the partial. Improve that text is not part of the scope of this PR.

ludrob commented on components/industry.spec.js:
```diff
@@ -0,0 +1,89 @@
+		const select = component.find('select#selectId');
+
+		expect(select.exists()).toBe(true);
+		expect(select.exists()).toBe(true);
```
```suggestion
```

ludrob commented on components/industry.jsx:
```diff
@@ -0,0 +1,56 @@
+const defaultOptions = demographics.industries.industries;
+
+export default function Industry ({
+	value = '',
```
one generic comment, when we have a list of prop, can we try to keep them in alphabetical order? I find it much quicker when I scan code for somehing

umbobabo replied:
The cognitive effort for developers to sort the properties alphabetically writing the code will surely overcome the scanning time we are going to gain. Happy to do it if you find an automated solution via TS or Prettier (https://palantir.github.io/tslint/rules/object-literal-sort-keys/).

Furthermore, these interfaces will be much simpler when we can refactor and create sub-components.

ludrob replied:
There is a sort line plugin for intellij and I think for vscode too.
I don't feel strongly about it, it's really just a suggestion,I find it more readable/logical, brings some order to the chaos :)

ludrob commented on components/job-title.spec.js:
```diff
@@ -0,0 +1,69 @@
+		const input = component.find('input#inputId');
+
+		expect(input.exists()).toBe(true);
+		expect(input.exists()).toBe(true);
```
dup
```suggestion
```

ludrob commented on partials/position.html:
```diff
@@ -1,16 +1,16 @@
 <div id="positionField" class="o-forms o-forms--wide ncf__field js-field{{#if hasError}} o-forms--error{{/if}}" data-ui-item="select" data-ui-item-name="position" data-validate="required">
 
-	<label for="position" class="o-forms__label">What's your job position?</label>
+	<label for="position" class="o-forms__label">What&#x27;s your job position?</label>
```
Again some weird quote, intended?
```suggestion
	<label for="position" class="o-forms__label">What's your job position?</label>
```

umbobabo replied:
Can't do this cause React convert the `'` into an entity (&#x27;), that is appropriate, and this will break the test.

ludrob commented on partials/job-title.html:
```diff
@@ -2,11 +2,11 @@
 				data-trackable="job-title"
 				aria-required="true" required
-				{{#if isDisabled}}disabled{{/if}}>
+				{{#if isDisabled}}disabled{{/if}} value="{{value}}">
```
```suggestion
				{{#if isDisabled}}disabled{{/if}} value="{{value}}"
	/>
```

ludrob commented on components/position.spec.js:
```diff
@@ -0,0 +1,82 @@
+
+		const select = component.find('select#selectId');
+		expect(select.exists()).toBe(true);
+		expect(select.exists()).toBe(true);
```
duplicate?
```suggestion
```

umbobabo commented:
@liamkeaton It breaks the client-side validation in Select cause if a value is not set explicit in the option, the value passed is the text inside the option tag so the validation now receives something like 'Please select your job title' and pass the CS validation as a string but fails on the SS validation. I will have a look at it tomorrow. FYI @dapenguin and @ludrob . This shouldn't apply for text input.

umbobabo commented:
@liamkeaton I changed the code just [here](https://github.com/Financial-Times/n-conversion-forms/pull/278/files#diff-356381f023b2a091fcad611705940c52R10) and [here](https://github.com/Financial-Times/n-conversion-forms/pull/278/files#diff-8a6bd21b5c8a917f62818915b17b3418R11). Sorry I rebased before you could review.

Next-subscribe client side validation is now working as expected

![Screenshot 2019-10-31 at 12 24 54](https://user-images.githubusercontent.com/752680/67955390-737cf980-fbea-11e9-8bdc-b2f5fd4cc0a9.png)

I think this I ready to go, please review.

liamkeaton commented on test/partials/industry.spec.js:
```diff
@@ -33,7 +33,6 @@ describe('industry template', () => {
 		const $ = context.template({});
 		expect($('select').find('option').length).to.equal(1);
 
-		expect($('select option').first().attr('value')).to.equal('');
```
Can we add these tests back in as now they should pass

liamkeaton commented on partials/position.html:
```diff
@@ -1,16 +1,16 @@
 <div id="positionField" class="o-forms o-forms--wide ncf__field js-field{{#if hasError}} o-forms--error{{/if}}" data-ui-item="select" data-ui-item-name="position" data-validate="required">
 
-	<label for="position" class="o-forms__label">What's your job position?</label>
+	<label for="position" class="o-forms__label">What&#x27;s your job position?</label>
```
Can we get to the bottom of why the `'` character is not allowed, it's a perfectly valid character is used in a number of our templates. It would be a shame to have to replace all of them with an unintelligible ascii code.

umbobabo replied:
It's this rule in Eslint-plugin https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-unescaped-entities.md#prevent-invalid-characters-from-appearing-in-markup-reactno-unescaped-entities that prevent me to use the character. This need then to be ported to HBS to have the same rendering.

The rule is sensible and probably we should keep it so to prevent undesired breakages and create awareness. I will disable the rule for this case cause we are sure it doesn't produce an undesired effect.

I pushed a change, have a look, please.

umbobabo replied:
In reality I think we should use `’` in this case rather than `'`.  In FT articles [like this](https://www.ft.com/content/eb5191ee-fbbf-11e9-a354-36acbbb0d9b6) the curly apostrophe is used rather then the single quote for copy like `...China’s growing military...`. I think it worth change this.

liamkeaton replied:
If that's what is used in other copy then it might be worth changing. Happy with this solution for now 👍
